### PR TITLE
Use is visible to determine if product is publish for 2 11

### PR DESCRIPTION
**Pull request by korycins**
Add resolvers for is_publish to calculate result based on db values - is_published and publication_date
It is cherry-picked changes from #6244 for 2.11

### CODE REVIEW

File: saleor/graphql/page/tests/test_page.py [hunk 1]
```diff
@@ -468,3 +469,64 @@ def test_query_pages_with_sort(
...
         assert pages[order]["node"]["title"] == page_name
+
+
+QUERY_PAGE_IS_PUBLISHED = """
+    query Page($id: ID!) {
+        page(id: $id) {
+            isPublished
+        }
+    }
+    """
```
Comment (tomaszszymanski129) [inline saleor/graphql/page/tests/test_page.py:480]:
```suggestion
"""
```

File: saleor/graphql/page/tests/test_page.py [hunk 2]
```diff
@@ -468,3 +469,64 @@ def test_query_pages_with_sort(
...
+
+    variables = {"id": graphene.Node.to_global_id("Page", page.pk)}
+    staff_api_client.user.user_permissions.add(permission_manage_pages)
+
+    with freeze_time(publication_date.replace(day=publication_date.day - 1)):
+        response = staff_api_client.post_graphql(QUERY_PAGE_IS_PUBLISHED, variables)
+        content = get_graphql_content(response, ignore_errors=True)
+        data = content["data"]["page"]
+        is_published = data["isPublished"]
+        assert is_published is False
```
Comment (tomaszszymanski129) [inline saleor/graphql/page/tests/test_page.py:500]:
```suggestion
        assert data["isPublished"] is False
```
could be shorter here and in few other places

File: saleor/graphql/product/tests/test_collection.py [hunk 3]
```diff
@@ -1004,3 +1009,66 @@ def test_sort_collection_products_by_name(staff_api_client, collection, product_
...
     ]
+
+
+QUERY_COLLECTION_IS_PUBLISHED = """
+    query Collection($id: ID!) {
+        collection(id: $id) {
+            isPublished
+        }
+    }
+    """
```
Comment (tomaszszymanski129) [inline saleor/graphql/product/tests/test_collection.py:1020]:
```suggestion
"""
```

File: saleor/graphql/product/tests/test_product.py [hunk 4]
```diff
@@ -1440,6 +1441,67 @@ def test_create_product(
     assert color_value_slug in values
 
 
+QUERY_PRODUCT_IS_PUBLISHED = """
+    query Product($id: ID!) {
+        product(id: $id) {
+            isPublished
+        }
+    }
+    """
```
Comment (tomaszszymanski129) [inline saleor/graphql/product/tests/test_product.py:1450]:
```suggestion
"""
```

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**